src/styles/_catalog.scss

```diff
@@ -1,22 +1,36 @@
-.catalog {
-  display: flex;
-  flex-wrap: wrap;
+$card-width: 200px;
 
-  max-width: 944px;
-  margin: 0 auto;
+.catalog {
+  display: grid;
+  justify-items: center;
+  justify-content: center;
+  gap: 48px;
 
   &__card {
-    display: flex;
+    display: grid;
     justify-content: center;
     align-items: center;
-    width: 200px;
+    width: $card-width;
     height: 300px;
     border: 1px solid #ccc;
     border-radius: 5px;
-    margin-bottom: 48px;
   }
+}
+
+@media (min-width: 488px) {
+  .catalog {
+    grid-template-columns: repeat(2, $card-width);
+  }
+}
+
+@media (min-width: 768px) {
+  .catalog {
+    grid-template-columns: repeat(3, $card-width);
+  }
+}
 
-  &__card:not(:nth-child(4n)) {
-    margin-right: 48px;
+@media (min-width: 1024px) {
+  .catalog {
+    grid-template-columns: repeat(4, $card-width);
```

Comment:
Also, one more trick - you could set the CSS variable for the number of cards and change in the media query.
Example:
```
--cards-number: 4;

.catalog {
  grid-template-columns: repeat(var(--cards-number), $card-width);
  
  @media (min-width: 488px) {
     --cards-number: 2;
  }
   /* and so on */
}
```
More info [here](https://www.w3schools.com/css/css3_variables_mediaqueries.asp)